digicampipe/image/utils.py

```diff
@@ -0,0 +1,416 @@
+import numpy as np
+import scipy
+from skimage.draw import polygon
+import matplotlib.pyplot as plt
+from time import clock
+import cv2
+from scipy import signal, ndimage, optimize
+import numbers
+from photutils import DAOStarFinder
+
+
+def crop_image(image, crop_pixel1, crop_pixel2):
+    """
+    :param image: 2D array of the image data with a shape = (vertical size, horizontal size)
+    :param crop_pixel1: iterable of 2 ints: position (pos_x , pos_y) of one corner of the returned region
+    :param crop_pixel2: iterable of 2 ints: position (pos_x , pos_y) of the opposite corner of the returned region
+    :return: the cropped image, the 1st corner and the opposite corner positions
+    """
+    if type(image) is str:
+        image = fits.open(image)[0].data
+    if type(image) is not np.ndarray:
+        raise AttributeError("image must be a filename or a numpy array.")
+    if crop_pixel1 is not list and len(crop_pixel1) is not 2:
+        raise AttributeError("crop_pixel1 must be a list of 2 ints", "(position in px of one corner of the crop area).")
+    if crop_pixel2 is not list and len(crop_pixel1) is not 2:
+        raise AttributeError("crop_pixel2 must be a list of 2 ints",
+                             "(position in px of the other corner of the crop area).")
+    crop_xmin, crop_ymin = crop_pixel1
+    crop_xmax, crop_ymax = crop_pixel2
+    if not isinstance(crop_xmin, numbers.Integral) or not isinstance(crop_ymin, numbers.Integral):
+        print('Warning in crop_image(): crop_pixel1 does not contain ints')
+        crop_xmin = int(crop_xmin)
+        crop_ymin = int(crop_ymin)
+    if not isinstance(crop_xmax, numbers.Integral) or not isinstance(crop_ymax, numbers.Integral):
+        print('Warning in crop_image(): crop_pixel2 does not contain ints')
+        crop_xmax = int(crop_xmax)
+        crop_ymax = int(crop_ymax)
+    npixel_y, npixel_x = image.shape
+    if not (0 <= crop_xmin < npixel_x and 0 <= crop_ymin < npixel_y):
+        raise AttributeError("invalid crop_pixel position:", crop_pixel1, crop_pixel2, "image size is: ", image.shape)
+    if not (0 < crop_xmax < npixel_x and 0 < crop_ymax < npixel_y):
+        raise AttributeError("invalid crop_pixel position:", crop_pixel1, crop_pixel2, "image size is: ", image.shape)
+    if crop_xmin > crop_xmax:
+        crop_xmin, crop_xmax = crop_xmax, crop_xmin
+    if crop_ymin > crop_ymax:
+        crop_ymin, crop_ymax = crop_ymax, crop_ymin
+    crop_pixel1 = crop_xmin, crop_ymin
+    crop_pixel2 = crop_xmax, crop_ymax
+    return image[crop_ymin:crop_ymax, crop_xmin:crop_xmax], crop_pixel1, crop_pixel2
+
+
+def average_images(images):
+    if type(images) is not list:
+        print('ERROR in average_images(): images must be a list of filenames or numpy.ndarray')
+        return None
+    nimage = len(images)
+    if nimage == 0:
+        print('ERROR in average_images(): empty list of images')
+        return None
+    if type(images[0]) is not np.ndarray:
```

Comment:
this checks, that the type of the first element is correct, but not the other elements. So it is pretty useless... 
Also later there is `images[0].shape` which will fail if `images[0]` does not have a property called `shape`...